rekt.go

```diff
@@ -47,63 +47,53 @@ func onMessageSent3(session *discordgo.Session, m *discordgo.MessageCreate) {
 	if len(content) < 5 {
 		return
 	}
-	var ban bool
-	if content[:5] == "kick " {
-		ban = false
-	} else {
-		if content[:4] == "ban " {
-			ban = true
-		} else {
-			return
+	if content[:2] == "i!" { // bot woke
+		var fields = strings.Fields(content)
+		var command = fields[0][2:]
+		if command == "kick" || command == "ban" || command == "mute" || command == "unmute" { // we don't want role checking outside of these commands
+			author, err := GetMember(msg.Author.ID)
+			if err != nil || !isStaff(author) {
+				return
+			}
+			if len(msg.Mentions) != 1 {
+				resp(msg.ChannelID, "Mention exactly one user")
+				return
+			}
+			user := msg.Mentions[0]
+			member, err := GetMember(user.ID)
+			if err != nil {
+				return
+			}
+			if len(member.Roles) > 0 {
+				resp(msg.ChannelID, "They have role(s)")
+				return
+			}
+			if !hasRole(author, STAFF["moderator"], STAFF["developer"]) && !evalRatelimit(msg.Author.ID) {
+				resp(msg.ChannelID, "Too soon")
+				return
+			}
+			providedReason := strings.TrimSpace(content[strings.Index(content, ">")+1:])
+			providedReason = command + " has been issued to " + user.Username + " by @" + msg.Author.Username + "#" + msg.Author.Discriminator + " for reason : " + providedReason
+			switch command {
+			case "ban":
+				err = discord.GuildBanCreateWithReason(m.GuildID, user.ID, providedReason, 0)
+				break
```

Comment:
Go [doesn't need](https://yourbasic.org/golang/switch-statement/) `break` in switch cases, if you wanted it to not break you'd need to explicitly include `fallthrough`

> The cases are evaluated from top to bottom, **stopping when a case succeeds**.